e2e_tests/tests/cluster/test_webhooks.py

```diff
@@ -125,27 +126,147 @@ def test_log_pattern_send_webhook(should_match: bool) -> None:
+    bindings.post_PostWebhook(
+        sess,
+        body=bindings.v1Webhook(
+            url=f"http://localhost:{port}{specific_path_unmatch}",
+            webhookType=bindings.v1WebhookType.DEFAULT,
+            triggers=[webhook_trigger],
+            mode=bindings.v1WebhookMode.SPECIFIC,
+            name=f"webhook-test{random.random()}",
+            workspaceId=1,
+        ),
+    )
+
     exp_id = exp.create_experiment(
         sess,
         conf.fixtures_path("no_op/single-medium-train-step.yaml"),
         conf.fixtures_path("no_op"),
-        ["--config", "hyperparameters.metrics_sigma=-1.0"],
+        [
+            "--config",
+            "hyperparameters.metrics_sigma=-1.0",
+            "--config",
+            "integrations.webhooks.webhook_name=['specific-webhook']",
+            "--project_id",
+            f"{project.id}",
+        ],
     )
     exp.wait_for_experiment_state(sess, exp_id, bindings.experimentv1State.ERROR)
 
-    for _ in range(10):
+    for _ in range(8):
         responses = server.return_responses()
-        if default_path in responses and slack_path in responses:
-            break
         time.sleep(1)
 
     responses = server.close_and_return_responses()
     if should_match:
-        assert len(responses) >= 2
+        assert len(responses) >= 3
         # Only need a spot check we get the default / slack responses.
         # Further tested in integrations.
         assert "TASK_LOG" in responses[default_path]
         assert "This log matched the regex" in responses[slack_path]
+        assert "TASK_LOG" in responses[default_path]
+        assert specific_path_unmatch not in responses
     else:
         assert default_path not in responses
         assert slack_path not in responses
+        assert specific_path not in responses
+        assert specific_path_unmatch not in responses
+
+
+@pytest.mark.e2e_cpu
+def test_specific_webhook() -> None:
+    port1 = 5007
+    server1 = utils.WebhookServer(port1, allow_dupes=True)
+    port2 = 5008
+    server2 = utils.WebhookServer(port2, allow_dupes=True)
+    sess = api_utils.admin_session()
+
+    workspace = bindings.post_PostWorkspace(
+        sess, body=bindings.v1PostWorkspaceRequest(name=f"webhook-test{random.random()}")
+    ).workspace
+    project = bindings.post_PostProject(
+        sess,
+        body=bindings.v1PostProjectRequest(
+            name=f"webhook-test{random.random()}",
+            workspaceId=workspace.id,
+        ),
+        workspaceId=workspace.id,
+    ).project
+
+    webhook_trigger = bindings.v1Trigger(
+        triggerType=bindings.v1TriggerType.EXPERIMENT_STATE_CHANGE,
+        condition={"state": "COMPLETED"},
+    )
+
+    webhook_1 = bindings.v1Webhook(
+        url=f"http://localhost:{port1}",
+        webhookType=bindings.v1WebhookType.SLACK,
+        triggers=[webhook_trigger],
+        mode=bindings.v1WebhookMode.SPECIFIC,
+        name=f"webhook_1{random.random()}",
+        workspaceId=1,
```

Comment:
Is this meant to be `1` or `workspace.id`?

Reply:
This is meant to be 1 so that this webhook is at a different workspace from the experiment, and the test is to show that this webhook does not get triggered.